From 91c395844c280f07b4eeb4aa20f1b2abd9fe1126 Mon Sep 17 00:00:00 2001
From: Torsten Veller <tove@gentoo.org>
Date: Sat, 14 Apr 2012 13:34:20 +0200
Subject: Provide a sensible INSTALLDIRS default for modules installed from
 CPAN.

Some modules which are included in core set INSTALLDIRS => 'perl'
explicitly in Makefile.PL or Build.PL.  This makes sense for the normal @INC
ordering, but not ours.
Taken from Debian.

Patch-Name: gentoo/cpan_definstalldirs.diff
---
 cpan/CPAN/lib/CPAN/FirstTime.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cpan/CPAN/lib/CPAN/FirstTime.pm b/cpan/CPAN/lib/CPAN/FirstTime.pm
index 5030ef9..5953cb7 100644
--- a/cpan/CPAN/lib/CPAN/FirstTime.pm
+++ b/cpan/CPAN/lib/CPAN/FirstTime.pm
@@ -990,7 +990,7 @@ sub init {
     my_prompt_loop(prefer_installer => 'MB', $matcher, 'MB|EUMM|RAND');
 
     if (!$matcher or 'makepl_arg make_arg' =~ /$matcher/) {
-        my_dflt_prompt(makepl_arg => "", $matcher);
+        my_dflt_prompt(makepl_arg => "INSTALLDIRS=site", $matcher);
         my_dflt_prompt(make_arg => "", $matcher);
         if ( $CPAN::Config->{makepl_arg} =~ /LIBS=|INC=/ ) {
             $CPAN::Frontend->mywarn(
@@ -1022,7 +1022,7 @@ sub init {
     my_dflt_prompt(make_install_arg => $CPAN::Config->{make_arg} || "",
                    $matcher);
 
-    my_dflt_prompt(mbuildpl_arg => "", $matcher);
+    my_dflt_prompt(mbuildpl_arg => "--installdirs site", $matcher);
     my_dflt_prompt(mbuild_arg => "", $matcher);
 
     if (exists $CPAN::HandleConfig::keys{mbuild_install_build_command}
